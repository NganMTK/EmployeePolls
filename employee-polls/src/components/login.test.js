import * as React from "react";
import {render, fireEvent } from "@testing-library/react";
import {Provider} from "react-redux";
import { BrowserRouter } from "react-router-dom";
import {createStore} from "redux";
import Login from "./login";
import recuder from "../reducers";
import middleware from "../middleware";

const store = createStore(recuder, middleware);

describe("Login", ()=> {
    it("have render component", ()=>{
        var component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );
        
        expect(component).toBeDefined();
        expect(component).toMatchSnapshot();
    })

    it("will have all expected fields", ()=>{
        var component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );
        
        var usernameInput = component.getByTestId("usename-input")
        var passwordInput = component.getByTestId("password-input")
        expect(usernameInput).toBeInTheDocument();
        expect(passwordInput).toBeInTheDocument();

        var submitButton = component.getByText("Submit")
        expect(submitButton).toBeInTheDocument();
    })

    it("will display alert error when input wrong username", ()=>{
        var component = render(
            <Provider store={store}>
                <BrowserRouter>
                    <Login/>
                </BrowserRouter>
            </Provider>
        );
        
        var input = component.getByTestId("usename-input");
        fireEvent.change(input, { target: { value: 'username' } });

        var submitButton = component.getByTestId("submit-button");
        fireEvent.click(submitButton);
        expect(component.getByTestId("alert-error")).toBeInTheDocument();
    })

    // it("will not display alert error when input correct username and password", ()=>{
    //     var component = render(
    //         <Provider store={store}>
    //             <BrowserRouter>
    //                 <Login/>
    //             </BrowserRouter>
    //         </Provider>
    //     );
        
    //     var usernameInput = component.getByTestId("usename-input");
    //     fireEvent.change(usernameInput, { target: { value: "sarahedo" } });
    //     var passwordInput = component.getByTestId("password-input");
    //     fireEvent.change(passwordInput, { target: { value: "password123" } });

    //     var submitButton = component.getByTestId("submit-button");
    //     fireEvent.click(submitButton);
    //     expect(component.queryByTestId("alert-error")).not.toBeInTheDocument();
    // })
})
