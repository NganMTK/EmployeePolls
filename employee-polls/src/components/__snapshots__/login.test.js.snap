// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login have render component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="login-container"
      >
        <div
          class="login-header"
        >
          <h2>
            Log In
          </h2>
        </div>
        <div
          class="login-field-group"
        >
          <div>
            <label
              class="login-username"
            >
              User
            </label>
            <input
              data-testid="usename-input"
              name="username"
              placeholder="Username"
              required=""
              type="text"
              value="sarahedo"
            />
          </div>
          <div>
            <label>
              Password
            </label>
            <input
              data-testid="password-input"
              name="password"
              placeholder="Password"
              required=""
              type="text"
              value="password123"
            />
          </div>
          <div
            class="login-submit-btn"
          >
            <input
              data-testid="submit-button"
              type="submit"
              value="Submit"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="login-container"
    >
      <div
        class="login-header"
      >
        <h2>
          Log In
        </h2>
      </div>
      <div
        class="login-field-group"
      >
        <div>
          <label
            class="login-username"
          >
            User
          </label>
          <input
            data-testid="usename-input"
            name="username"
            placeholder="Username"
            required=""
            type="text"
            value="sarahedo"
          />
        </div>
        <div>
          <label>
            Password
          </label>
          <input
            data-testid="password-input"
            name="password"
            placeholder="Password"
            required=""
            type="text"
            value="password123"
          />
        </div>
        <div
          class="login-submit-btn"
        >
          <input
            data-testid="submit-button"
            type="submit"
            value="Submit"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
